File: Data/CounterData.cs
using System;

namespace Data
{
    public class CounterData
    {
        public int Id { get; set; }

        public double Reading { get; set; }

        public double? ReadingODN { get; set; }

        public DateTime CreatedDate { get; set; }

        public bool IsFirst { get; set; }


        public int CounterId { get; set; }

        public virtual Counter Counter { get; set; }


        public int CounterTarifId { get; set; }

        public virtual CounterTarif CounterTarif { get; set; }
    }
}

File: Comunal/app/models/counterData.ts
﻿export class CounterData {
    Id: number;
    TarifCount: number;

    Tarif1: number;
    Limit1: number;
    Tarif2: number;
    Limit2: number;
    Tarif3: number;
    IconPath: string;
    LastReading: number;
    LastReadingDate: any;
    EnableODN: boolean;
    LastCounterDataDTO: CounterData;
    StartReading: number;

    constructor(
        public Reading: number = 0,
        public ReadingODN: number = null,
        public ReadingDateDay: number = null,
        public ReadingDateYear: number = null,
        public ReadingDateMonth: number = null,
        public IsFirst: boolean = false,

        public CounterName: string = null,
        public CounterId: number = null,
        public CounterTypeId: number = null,
        public BillId: number = null,
        public CounterTarifId: number = null
        ) { }
}